xbmc/cores/AudioEngine/Sinks/pipewire/Pipewire.cpp

```diff
@@ -30,10 +30,7 @@ CPipewire::CPipewire()
 CPipewire::~CPipewire()
 {
   if (m_loop)
-  {
-    m_loop->Unlock();
```

Comment:
The problem here is though, that the ctx and the pipewire object share the mainloop. While calling Unlock might be wrong - as ctx currently does not hold the lock, but assume for a moment it did, then you would kill the ctx object under the hood of pw processing something.

See in comparison to the ::Start method and how this is encapsulated in between: The ScopeGuard.
I would scope-guard the destructor as well if m_loop is still a real thing. That way it would also not race towards resetting / stopping of ctx. The design is a bit unclean as it mixes mainloop as protection for lifetime of an object and internal forth and back.

Does that work for you:
if (m_loop)
{
    PIPEWIRE::CLoopLockGuard lock(*m_loop);
    m_loop->Stop();
    m_context.reset(); // shares the lock, make sure it does not act, so protect with lock
  }
  else 
  {
        m_context.reset(); // needs to be done also in else case, but shares loop with other guy
  }
    m_loop.reset(); // Would  call the loop destory pw method if it existed, so lock needs unhold here (scopeguard ends).
    pw_deinit();
  }